### PR TITLE
Don't die if folder doesn't have id

### PR DESCRIPTION
**Pull request by crobibero**
image

-- 
Seems to be from a (now) non-existent library path

### CODE REVIEW

**Comment by hawken93** [comment]
Worthy of note: The server which experienced this probably had its libraries moved to a different location and therefore couldn't find them on disk. I'll get back to you when I know if it helped to have the libraries in the same location as before.

**Review by cvium — APPROVED** [review]

File: Jellyfin.Server/Migrations/Routines/RemoveDownloadImagesInAdvance.cs [hunk 1]
```diff
@@ -35,8 +35,14 @@ public void Perform()
...
             {
+                // Some virtual folders don't have a proper item id.
+                if (!Guid.TryParse(virtualFolder.ItemId, out var folderId))
+                {
+                    continue;
+                }
+
                 var libraryOptions = virtualFolder.LibraryOptions;
-                var collectionFolder = (CollectionFolder)_libraryManager.GetItemById(virtualFolder.ItemId);
+                var collectionFolder = (CollectionFolder)_libraryManager.GetItemById(folderId);
```
Comment (cvium) [inline Jellyfin.Server/Migrations/Routines/RemoveDownloadImagesInAdvance.cs:45]: A thought occurred to me. Playlists and Collections are also virtual folders, right? Do they have library options?

Comment (ferferga) [reply Jellyfin.Server/Migrations/Routines/RemoveDownloadImagesInAdvance.cs:45]: @cvium The migration ran fine for me in a server with playlists and collections.

Comment (cvium) [reply Jellyfin.Server/Migrations/Routines/RemoveDownloadImagesInAdvance.cs:45]: I don't doubt that, I just wonder if it creates library options for non libraries 🤔

Comment (ferferga) [reply Jellyfin.Server/Migrations/Routines/RemoveDownloadImagesInAdvance.cs:45]: @cvium I can do an stable-> unstable upgrade again.

I'll check once I arrive home and I'll report back

Comment (ferferga) [reply Jellyfin.Server/Migrations/Routines/RemoveDownloadImagesInAdvance.cs:45]: @cvium Copied data from stable and re-ran the migration: only what's seen in the Dashboard > Libraries section is treated here. This includes the "Collections" library that gets created when you create a collection **but not the collections themselves**. All my playlists are also unaffected.

**Review by nielsvanvelzen — APPROVED** [review]

**Pull request merged**